core/src/test/java/io/lionweb/lioncore/java/serialization/MyNodeWithProperties.java

```diff
@@ -33,8 +33,9 @@ public MyNodeWithProperties(String id) {
     super(id, CONCEPT);
   }
 
-  public boolean getP1() {
-    return (boolean) this.getPropertyValueByName("p1");
+  public Boolean getP1() {
+    Object value = this.getPropertyValueByName("p1");
+    return value == null ? null : (Boolean) value;
```

Comment:
Is this check actually needed? I think we can cast `null` to `Boolean`

Reply:
I will check that. It is such a long time that I do not use Java!

Reply:
You are right, I fixed that in master as I had already merged this